exchanges/coinbasepro/coinbasepro_test.go

```diff
@@ -2,1074 +2,1952 @@ package coinbasepro
+		kline.TwoHour:    granTwoHour,
+		kline.SixHour:    granSixHour,
+		kline.OneDay:     granOneDay,
+		kline.OneWeek:    errIntervalNotSupported}
+	for k := range testSequence {
+		resp := FormatExchangeKlineIntervalV3(k)
+		if resp != testSequence[k] {
+			t.Errorf(errExpectMismatch, resp, testSequence[k])
+		}
 	}
 }
 
-func TestWithdrawInternationalBank(t *testing.T) {
+func TestGetCurrencyTradeURL(t *testing.T) {
 	t.Parallel()
-	sharedtestvalues.SkipTestIfCannotManipulateOrders(t, c, canManipulateRealOrders)
-
-	var withdrawFiatRequest = withdraw.Request{
-		Amount:   100,
-		Currency: currency.USD,
-		Fiat: withdraw.FiatRequest{
-			Bank: banking.Account{
-				BankName: "Federal Reserve Bank",
-			},
-		},
+	testexch.UpdatePairsOnce(t, c)
+	for _, a := range c.GetAssetTypes(false) {
+		pairs, err := c.CurrencyPairs.GetPairs(a, false)
+		require.NoError(t, err, "cannot get pairs for %s", a)
+		require.NotEmpty(t, pairs, "no pairs for %s", a)
+		resp, err := c.GetCurrencyTradeURL(context.Background(), a, pairs[0])
+		require.NoError(t, err)
+		assert.NotEmpty(t, resp)
 	}
+}
 
-	_, err := c.WithdrawFiatFundsToInternationalBank(context.Background(),
-		&withdrawFiatRequest)
-	if !sharedtestvalues.AreAPICredentialsSet(c) && err == nil {
-		t.Error("Expecting an error when no keys are set")
+func TestScheduleFuturesSweep(t *testing.T) {
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, c, canManipulateRealOrders)
+	curSweeps, err := c.ListFuturesSweeps(context.Background())
+	assert.NoError(t, err)
+	preCancel := false
+	if len(curSweeps) > 0 {
+		for i := range curSweeps {
+			if curSweeps[i].Status == "PENDING" {
+				preCancel = true
+			}
+		}
 	}
-	if sharedtestvalues.AreAPICredentialsSet(c) && err != nil {
-		t.Errorf("Withdraw failed to be placed: %v", err)
+	if preCancel {
+		_, err = c.CancelPendingFuturesSweep(context.Background())
+		assert.NoError(t, err)
 	}
+	_, err = c.ScheduleFuturesSweep(context.Background(), 0.001337)
+	assert.NoError(t, err)
 }
 
-func TestGetDepositAddress(t *testing.T) {
-	_, err := c.GetDepositAddress(context.Background(), currency.BTC, "", "")
-	if err == nil {
-		t.Error("GetDepositAddress() error", err)
+func TestCancelPendingFuturesSweep(t *testing.T) {
+	sharedtestvalues.SkipTestIfCredentialsUnset(t, c, canManipulateRealOrders)
+	curSweeps, err := c.ListFuturesSweeps(context.Background())
+	assert.NoError(t, err)
+	partialSkip := false
+	if len(curSweeps) > 0 {
+		for i := range curSweeps {
+			if curSweeps[i].Status == "PENDING" {
+				partialSkip = true
+			}
+		}
+	}
+	if !partialSkip {
+		_, err = c.ScheduleFuturesSweep(context.Background(), 0.001337)
+		assert.NoError(t, err)
 	}
+	_, err = c.CancelPendingFuturesSweep(context.Background())
+	assert.NoError(t, err)
 }
 
 // TestWsAuth dials websocket, sends login request.
 func TestWsAuth(t *testing.T) {
-	if !c.Websocket.IsEnabled() && !c.API.AuthenticatedWebsocketSupport || !sharedtestvalues.AreAPICredentialsSet(c) {
+	// t.Parallel()
```

Comment:
reinstate?

Reply:
Just gonna rm that, since other tests interfere with it.